### ISSUE TITLE
HELP: Receiving Errors while trying to use web-extension-polyfill-ts 

### ISSUE THREAD

**Issue by daveavi** [Lusito/webextension-polyfill-ts#61]
### Overview
I have been building a chrome extension for a while using the chrome api's, but I want to make it compatible with Firefox as well.  I have just discovered this wrapper and I am trying to experiment with it to see if it works. I am having an issue where if I switch from using the chrome API to using web-extenion-polyfill in my background script, I receive an error that looks like this: 

<img width="657" alt="Screen Shot 2021-06-19 at 12 56 38 PM" src="https://user-images.githubusercontent.com/20130700/122649832-dbed5700-d0fd-11eb-827f-0401e5dfb823.png">

Right now, my background script looks something like this: 
<img width="816" alt="Screen Shot 2021-06-19 at 12 58 32 PM" src="https://user-images.githubusercontent.com/20130700/122649884-1951e480-d0fe-11eb-8be7-091096443b92.png">

If I were to uncomment the code that using the chrome API and comment the web-extension-polyfill-code, then I do not have any errors.

I was also thinking that at the moment I am using Vite to build my extension, and in my vite.config.ts, I have not specified anything about web-extension-polyfill-ts in the build process, I don't think that could be an issue here, but I here is my vite.config.ts:

<img width="906" alt="Screen Shot 2021-06-19 at 1 08 03 PM" src="https://user-images.githubusercontent.com/20130700/122650181-6aaea380-d0ff-11eb-91ca-0833930d288f.png"> 

I am a little new to Vite and using TS, so I know there could be something I am doing wrong, but help is much appreciated!

**Comment by dmchoull** [comment]
I believe the issue is that you are trying to use the browserAction API in a manifest version 3 extension. Chrome has changed that API to "action" in v3, as in your commented out code. This polyfill does not support "action" yet (see https://github.com/Lusito/webextension-polyfill-ts/issues/51 and the linked bugzilla bugs). It seems like Firefox will soon support action as an alias to browserAction, but in the meantime you will probably have to resort to doing something like this:

```typescript
// @ts-ignore
const action = browser.browserAction || browser.action
```

**Comment by Lusito** [comment]
I think @dmchoull is probably right. The ts-ignore should not be required with 0.26.0 anymore though.

If it's not fixed like that, try to show me the generated javascript code or, even better, try to find out what the actual error message is, that chrome is showing you.

**Comment by daveavi** [comment]
Hi, sorry but I had left this issue for a couple of weeks due to re-scoping of my project. I am experimenting with vite again, and what I recognized was that vite is bundling my ts file as a module, so I had to create a background.html page with a script tag that specifies that my background.js is a module. Although that error went away by doing this, I am having issues using my other ts files that I want to use. I am currently trying to inject a js file into a tab from my background script when I click on my extension logo:

<img width="890" alt="Screen Shot 2021-07-12 at 12 51 31 PM" src="https://user-images.githubusercontent.com/20130700/125326491-1bc7e880-e310-11eb-84f7-1f7a9666c73f.png">

Here is the script I am injecting:
<img width="1032" alt="Screen Shot 2021-07-12 at 12 51 39 PM" src="https://user-images.githubusercontent.com/20130700/125326721-56318580-e310-11eb-95d2-5fd92634fbfa.png">


When I click on the extension logo and try to send a message back to my background script I get the following error:
<img width="544" alt="Screen Shot 2021-07-12 at 12 52 07 PM" src="https://user-images.githubusercontent.com/20130700/125326743-5c276680-e310-11eb-96f7-4b117cdc8ec6.png">
 
And here is my vite.config file:
<img width="1046" alt="Screen Shot 2021-07-12 at 12 51 55 PM" src="https://user-images.githubusercontent.com/20130700/125326824-782b0800-e310-11eb-8c1c-d568159f0ffd.png">

The only way to solve this issue is making html files with adding script tags that specfiy the js file is a module. Is there a way to bundle our ts files as common JS scripts instead of modules?
